File: javascript/claude/p1.js
// Task 1: Function copies a number of bytes from one memory location to another
function copyBytes(dest, src, numBytes) {
  /**
   * Copies numBytes from src to dest.
   * dest: destination array
   * src: source array
   * numBytes: number of bytes to copy
   */
  for (let i = 0; i < numBytes; i++) {
    dest[i] = src[i];
  }
}

// Example usage
const src = Array.from("Hello, World!".split("").map((c) => c.charCodeAt(0)));
const dest = new Array(20).fill(0);

copyBytes(dest, src, 13);
console.log("Copied:", String.fromCharCode(...dest.slice(0, 13)));

File: python/claude/p1.py
# Task 1: Function copies a number of bytes from one memory location to another
def copy_bytes(dest, src, num_bytes):
    """
    Copies num_bytes from src to dest.
    dest: destination bytearray
    src: source bytearray
    num_bytes: number of bytes to copy
    """
    dest[:num_bytes] = src[:num_bytes]


# Example usage
if __name__ == "__main__":
    src = bytearray(b"Hello, World!")
    dest = bytearray(20)

    copy_bytes(dest, src, 13)
    print("Copied:", dest[:13].decode())
